lib/internal/modules/cjs/loader.js

```diff
@@ -1246,6 +1245,23 @@ Module._extensions['.node'] = function(module, filename) {
   return process.dlopen(module, path.toNamespacedPath(filename));
 };
 
+Module._extensions['.mjs'] = function(module, filename) {
+  const ESMLoader = asyncESM.ESMLoader;
+  const url = `${pathToFileURL(filename)}`;
+  const job = ESMLoader.getModuleJobWorker(url, 'module');
```

Comment:
so this just skips the esm resolver entirely? this seems like a rather dangerous divergence.

Reply:
That's because we already resolved the specifier, in order to know we needed the esm loader. The cjs resolver and the esm resolver are _supposed_ to resolve the same specifier to the same thing, so it _should_ be fine. This is part of why I kept saying it _really should only be considered to be one loader_.